code/modules/paperwork/paperbin.dm

```diff
@@ -120,50 +144,92 @@
 
 /obj/item/paper_bin/update_overlays()
 	. = ..()
+
+	total_paper = LAZYLEN(papers)
+
+	if(bin_pen)
+		pen_overlay = mutable_appearance(bin_pen.icon, bin_pen.icon_state)
+
+	if(!bin_overlay)
+		bin_overlay = mutable_appearance(icon, bin_overlay_string)
+
+	if(LAZYLEN(papers))
+		var/paper_number = 1
+		for(var/obj/item/paper/current_paper in papers)
+			if(paper_number == LAZYLEN(papers) || paper_number % 8 == 0)
+				var/mutable_appearance/paper_overlay = mutable_appearance(current_paper.icon, current_paper.icon_state)
+				paper_overlay.color = current_paper.color
+				paper_overlay.pixel_y = paper_number/8 - 2 //gives the illusion of stacking
+				. += paper_overlay
+				if(paper_number == LAZYLEN(papers)) //this is our top paper
+					. += current_paper.overlays //add overlays only for top paper
+					if(istype(src, /obj/item/paper_bin/bundlenatural))
+						bin_overlay.pixel_y = paper_overlay.pixel_y //keeps binding centred on stack
+					if(bin_pen)
+						pen_overlay.pixel_y = paper_overlay.pixel_y //keeps pen on top of stack
+			paper_number++
```

Comment:
Could you use continue to make this less indented?

Reply:
Did it, thanks